File: c/MoveAllZerosToEnd.c
#include <stdio.h> 

void func(int arr[], int n) { 
	int ctr = 0; 

	for (int i = 0; i < n; i++) 
		if (arr[i] != 0) 
			arr[ctr++] = arr[i];								

	while (ctr < n) 
		arr[ctr++] = 0; 
} 

int main() { 
	int arr[] = {1, 2, 3, 0, 0, 0, 10, 9, 8, 0, 5, 0}; 
	int n = 12; 
	func(arr, n); 

	for (int i = 0; i < n; i++) 
		printf("%d",arr[i]); 

	return 0; 
} 

File: python/MoveAllZerosToEnd.py
def end(arr, n): 
	ctr = 0 
	
	for i in range(n): 
		if arr[i] != 0: 
			
			arr[ctr] = arr[i] 
			ctr+=1
	
	while ctr < n: 
		arr[ctr] = 0
		ctr += 1
		
arr = [1, 2, 3, 0, 0, 0, 10, 9, 8, 0, 5, 0] 
n = len(arr) 
end(arr, n) 


